### PR TITLE
feat: skip closer pieces preference in sequential mode

### PR DESCRIPTION
**Pull request by G-Ray**
Following [this comment](https://github.com/transmission/transmission/pull/7949/changes#r3121908796), I open this standalone PR for this optimization to not prioritize pieces closer to completion in sequential mode.

Notes: Improved sequential downloading.

### CODE REVIEW

File: tests/libtransmission/peer-mgr-wishlist-test.cc [hunk 1]
```diff
@@ -222,65 +222,67 @@ TEST_F(PeerMgrWishlistTest, doesNotRequestSameBlockTwice)
 
 TEST_F(PeerMgrWishlistTest, sequentialDownload)
 {
-    auto const get_spans = [](size_t n_wanted)
```
Comment (tearfur) [inline tests/libtransmission/peer-mgr-wishlist-test.cc:225]: Please keep the `get_spans()` lambda and create a new Wishlist for each run (like it did before).

The idea is to fuzz out any edge cases that can be triggered by random elements. Although there is no random element in the wishlist in sequential mode *in the current implementation*, I'd like the tests to be future proof in case we ever decide to add a random element.

**Review by tearfur — CHANGES_REQUESTED** [review]
Wishlist changes look good. Just want a little change in the test code.

**Review by tearfur — APPROVED** [review]
